proto/types.proto

```diff
@@ -8,10 +8,14 @@ package types;
 
 message Tx {
   bytes code = 1;
-  optional bytes data = 2;
+  TxData data = 2;
```

Comment:
Is this change necessary? I think it would be good to keep this explicitly `optional`

Reply:
I tried building this with `optional`, but `protoc` couldn't build that. In my investigation, Protobuf v3.15 supports `optional`, `prost` which supports it hasn't released yet. I'm not sure why we couldn't build it directly, even though we could do it with `services.proto`.
https://github.com/danburkert/prost/issues/351

Reply:
It seems that this might be possible by setting `.protoc_arg("--experimental_allow_proto3_optional");` on the tonic build config. Would you mind to try this pls?

Reply:
That works well! Thank you! https://github.com/anomanetwork/anoma/pull/291/commits/2b4100e676ef22df69f213ce0c47ab97c0718ca2

Reply:
thank you too!